### PR TITLE
Fix server group disabled status and instance load balancer health checks

### PR DESCRIPTION
**Pull request by drmaas**

### CODE REVIEW

**Comment by drmaas** [comment]
@spinnaker/reviewers 
@derekolk 
@varikin

**Comment by derekolk** [comment]
Squash commits ... LGTM.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy [hunk 1]
```diff
@@ -238,4 +238,13 @@ interface OpenstackLoadBalancerProvider {
    * @return
    */
   LoadBalancerV2StatusTree getLoadBalancerStatusTree(final String region, final String id)
+
+  /**
+   * Updates a pool member status.
+   * @param region
+   * @param poolId
+   * @param memberId
+   * @param status true means the pool member is enabled, false means it's disabled and can recieve no traffic
```
Comment (duftler) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy]: s/recieve/receive

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/AbstractEnableDisableOpenstackAtomicOperation.groovy [hunk 2]
```diff
@@ -54,16 +57,16 @@ abstract class AbstractEnableDisableOpenstackAtomicOperation implements AtomicOp
   @Override
   Void operate(List priorOutputs) {
     String verb = disable ? 'disable' : 'enable'
-    String presentParticipling = disable ? 'Disabling' : 'Enabling'
+    String gerund = disable ? 'Disabling' : 'Enabling'
```
Comment (duftler) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/AbstractEnableDisableOpenstackAtomicOperation.groovy]: Nice.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackInstance.groovy [hunk 3]
```diff
@@ -16,64 +16,120 @@
...
+
+      // load balancer health
+      loadBalancerHealths?.each {
+        healths << mapper.convertValue(it.view, OpenstackInfrastructureProvider.ATTRIBUTES)
+      }
+
+      //instance health
+      healths << mapper.convertValue(instanceHealth?.view, OpenstackInfrastructureProvider.ATTRIBUTES)
+
+      //TODO - Add consul health
```
Comment (duftler) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackInstance.groovy]: Please include a user in the `TODO`.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackInstance.groovy [hunk 4]
```diff
@@ -16,64 +16,120 @@
...
+      def allHealths = []
+
+      loadBalancerHealths?.each {
+        allHealths << it.view
+      }
+      if (instanceHealth) {
+        allHealths << instanceHealth.view
+      }
+
+      //TODO - Add consul health views
```
Comment (duftler) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackInstance.groovy]: Same here.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackLoadBalancerCachingAgent.groovy [hunk 5]
```diff
@@ -31,33 +31,46 @@ import com.netflix.spinnaker.clouddriver.openstack.cache.Keys
...
+import org.openstack4j.model.network.ext.status.MemberV2Status
 
 import java.util.concurrent.CompletableFuture
 import java.util.concurrent.Future
 
 import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.AUTHORITATIVE
+import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.INFORMATIVE
 import static com.netflix.spinnaker.clouddriver.cache.OnDemandAgent.OnDemandType.LoadBalancer
 import static com.netflix.spinnaker.clouddriver.openstack.OpenstackCloudProvider.ID
+import static com.netflix.spinnaker.clouddriver.openstack.cache.Keys.Namespace.INSTANCES
```
Comment (duftler) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackLoadBalancerCachingAgent.groovy]: Sort-order is off here.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackLoadBalancerCachingAgent.groovy [hunk 6]
```diff
@@ -31,33 +31,46 @@ import com.netflix.spinnaker.clouddriver.openstack.cache.Keys
...
 import static com.netflix.spinnaker.clouddriver.openstack.cache.Keys.Namespace.FLOATING_IPS
 import static com.netflix.spinnaker.clouddriver.openstack.cache.Keys.Namespace.LOAD_BALANCERS
 import static com.netflix.spinnaker.clouddriver.openstack.cache.Keys.Namespace.NETWORKS
 import static com.netflix.spinnaker.clouddriver.openstack.cache.Keys.Namespace.SUBNETS
 import static com.netflix.spinnaker.clouddriver.openstack.provider.OpenstackInfrastructureProvider.ATTRIBUTES
 
+/*
+  TODO we could cache the load balancer status tree with each load balancer too, which would be used
```
Comment (duftler) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackLoadBalancerCachingAgent.groovy]: Please add a user id to the `TODO`.

**Pull request merged**